src/reseeding.rs

```diff
@@ -121,69 +176,36 @@ impl<R: RngCore + SeedableRng, Rsdr: RngCore> ReseedingRng<R, Rsdr> {
             Ok(())
         }
     }
-}
-
-impl<R: RngCore + SeedableRng, Rsdr: RngCore> RngCore for ReseedingRng<R, Rsdr> {
-    fn next_u32(&mut self) -> u32 {
-        let value = self.rng.next_u32();
-        self.bytes_until_reseed -= 4;
-        if self.bytes_until_reseed <= 0 {
-            self.reseed();
-        }
-        value
-    }
-
-    fn next_u64(&mut self) -> u64 {
-        let value = self.rng.next_u64();
-        self.bytes_until_reseed -= 8;
-        if self.bytes_until_reseed <= 0 {
-            self.reseed();
-        }
-        value
-    }
-
-    fn fill_bytes(&mut self, dest: &mut [u8]) {
-        self.rng.fill_bytes(dest);
-        self.bytes_until_reseed -= dest.len() as i64;
-        if self.bytes_until_reseed <= 0 {
-            self.reseed();
-        }
-    }
 
-    fn try_fill_bytes(&mut self, dest: &mut [u8]) -> Result<(), Error> {
-        let res1 = self.rng.try_fill_bytes(dest);
-        self.bytes_until_reseed -= dest.len() as i64;
-        let res2 = if self.bytes_until_reseed <= 0 {
-            self.try_reseed()
-        } else { Ok(()) };
-        
-        if let Err(e) = res1 {
-            // In the unlikely event the internal PRNG fails, we don't know
-            // whether this is resolvable; reseed immediately and return
-            // original error kind.
-            self.bytes_until_reseed = 0;
-            Err(e)
-        } else {
-            res2
-        }
+    #[inline(never)]
+    fn reseed_and_generate(&mut self,
+                           results: &mut <Self as BlockRngCore<u32>>::Results)
+        -> Result<(), Error>
+    {
+        let res1 = self.auto_reseed();
+        self.bytes_until_reseed -= results.as_ref().len() as i64 * 4;
+        let res2 = self.core.generate(results);
+        if res2.is_err() { res2 } else { res1 }
```

Comment:
Since `reseed_and_generate` is only used within `generate`, I suggest ignoring any error from `auto_reseed` here, but passing on errors from `generate` — because if reseeding fails we can still output fresh random numbers, but if the latter fails (if that's even possible) then we have a much more serious problem.

This implies that if `generate` doesn't return an `Result` then this shouldn't either; I'm not sure which way it should go, but IMO `generate` should never return errors from reseeding because there's nothing useful to do with them and it just ends up with bad handling like I already commented on above.